packages/controllers/src/permissions/Permission.ts

```diff
@@ -419,13 +504,17 @@ type PermissionSpecificationBuilderOptions<
 
 /**
  * A function that builds a permission specification. Modules that specify
- * restricted methods for external consumption should make this their primary /
+ * permissions for external consumption should make this their primary /
  * default export so that host applications can use them to generate concrete
  * specifications tailored to their requirements.
  */
 export type PermissionSpecificationBuilder<
+  Type extends PermissionType,
   Options extends PermissionSpecificationBuilderOptions<any, any, any>,
-  Specification extends PermissionSpecificationConstraint,
+  Specification extends Extract<
+    PermissionSpecificationConstraint,
+    { permissionType: Type }
+  >,
```

Comment:
I don't follow the purpose of this `Extract`. Functionally this is the same as

```suggestion
  Specification extends { permissionType: Type },
```

Which is a little odd. That's not much of a specification.

Reply:
Oh wait, I guess you meant this

```suggestion
  Specification extends PermissionSpecificationConstraint & {
    permissionType: Type
  },
```

Reply:
lol, yes!

Reply:
Fixed this and the other instances of this "pattern" here: 71b36f1